worlds/dlcquest/Rules.py

```diff
@@ -321,7 +330,7 @@ def has_coin(state, player: int, coins: int):
                                              math.ceil(25 / World_Options[Options.CoinSanityRange])))
 
         if World_Options[Options.CoinSanity] == Options.CoinSanity.option_none:
-            set_rule(world.get_entrance("Boss Door", player),
+            add_rule(world.get_entrance("Boss Door", player),
```

Comment:
I'm pretty sure this logic is wrong.
If you had a unit test for it, we could know for sure.
But from how I'm reading it, it seems like if you're on coinsanity none, the rule being added here used to overwrite the coinsanity rule added above, but now, it's not overwriting but being added to it. So on coinsanity none, the coinsanity rule will still be applied, leading to things not working as intended at all.

The obvious fix would be to addd the coinsanity boss door rule inside a coinsanity if, same as this one that is in the opposite if condition

Reply:
Revolved, turns out it was fine